msmtools/analysis/dense/gpcca.py

```diff
@@ -1060,59 +1076,58 @@ class GPCCA(object):
 
     """
 
-    def __init__(self, P, eta, z='LM', method='brandts'):
+    def __init__(self, P, eta=None, z='LM', method='brandts'):
         if issparse(P):
-            warnings.warn("G-PCCA is only implemented for dense matrices, "
-                          + "converting sparse transition matrix to dense ndarray.")
-            P = P.toarray()
+            if method != 'krylov':
+                raise ValueError("Sparse implementation is only available for `method='krylov'`.")
         from msmtools.analysis import is_transition_matrix
+
         if not is_transition_matrix(P):
             raise ValueError("Input matrix P is not a transition matrix.")
+        if z not in ['LM', 'LR']:
+            raise ValueError("You didn't give a valid sorting criterion z. Valid options are `'LM'` and `'LR'`.")
+        if method not in ['brandts', 'scipy', 'krylov']:
+            raise ValueError("You didn't give a valid method to determine the invariant subspace.")
+
         self.P = P
-        self.eta = eta
+        self.eta = (np.ones(P.shape[0]) / P.shape[0]) if eta is None else eta
```

Comment:
```suggestion
        if eta == None:
             self.eta = np.true_divide(np.ones(P.shape[0]), P.shape[0])
        else:
            self.eta = eta
```
I think this from adds some clarity. Also we need to keep in mind that / isn't doing true division in python 2.